types/signed_header.go

```diff
@@ -93,6 +92,31 @@ var (
 	ErrSignatureVerificationFailed = errors.New("signature verification failed")
 )
 
+// VerifyCentralizedSequencer checks if a SignedHeader came from the expected centralized sequencer, given a genesis file.
+func (sh *SignedHeader) VerifyCentralizedSequencer(genesis *cmtypes.GenesisDoc) error {
+	if len(genesis.Validators) != 1 {
+		return errors.New("provided genesis must have 1 validator (the centralized sequencer)")
+	}
+	if !bytes.Equal(sh.Validators.Proposer.PubKey.Bytes(), genesis.Validators[0].PubKey.Bytes()) {
+		return fmt.Errorf("signed header proposer public key %x does not match the centralized sequencer public key %x specified in the genesis file", sh.Validators.Proposer.PubKey.Bytes(), genesis.Validators[0].PubKey.Bytes())
```

Comment:
long error message is not required as it is not utilized in the callsite. since the callsite simply wants to skip over the junk, you can just do with `bool` which is the result of bytes compare. no need for additional error.